app/src/androidTest/java/com/example/mareu/MeetingAddTest.java

```diff
@@ -0,0 +1,128 @@
+import java.util.List;
+
+@RunWith(AndroidJUnit4.class)
+public class MeetingAddTest {
+
+    private MeetingRepository meetingRepository;
+
+    @Rule
+    public ActivityScenarioRule<MeetingListActivity> activityScenarioRule = new ActivityScenarioRule<MeetingListActivity> (MeetingListActivity.class);
+
+    @Before
+    public void setUp() {
+        meetingRepository = MeetingRepository.getInstance();
+    }
+
+    @Test
+    public void meetingAddTest_check() {
+        // Given
+        Meeting expectedMeeting = getFakeMeeting();
+
+        // When
+        onView(ViewMatchers.withId(R.id.add_meeting_button)).perform(click());
+
+        onView(ViewMatchers.withId(R.id.add_meeting_name))
+                .perform(replaceText(expectedMeeting.getName()));
+
+        onView(ViewMatchers.withId(R.id.add_meeting_date_button)).perform(click());
+        onView(ViewMatchers.withId(R.id.dialog_date_picker))
+                .perform(PickerActions.setDate(expectedMeeting.getBeginDate().get(Calendar.YEAR), expectedMeeting.getBeginDate().get(Calendar.MONTH), expectedMeeting.getBeginDate().get(Calendar.DAY_OF_MONTH)));
+        onView(ViewMatchers.withId(R.id.dialog_date_button)).perform(click());
+
+        onView(ViewMatchers.withId(R.id.add_meeting_hour_button)).perform(click());
+        onView(ViewMatchers.withId(R.id.dialog_hour))
+                .perform(PickerActions.setTime(expectedMeeting.getBeginDate().get(Calendar.HOUR_OF_DAY), expectedMeeting.getBeginDate().get(Calendar.MINUTE)));
+        onView(ViewMatchers.withId(R.id.dialog_hour_button)).perform(click());
+
+        onView(ViewMatchers.withId(R.id.add_meeting_endTime_button)).perform(click());
+        onView(ViewMatchers.withId(R.id.dialog_hour))
+                .perform(PickerActions.setTime(expectedMeeting.getEndDate().get(Calendar.HOUR_OF_DAY), expectedMeeting.getEndDate().get(Calendar.MINUTE)));
+        onView(ViewMatchers.withId(R.id.dialog_hour_button)).perform(click());
+
+        onView(ViewMatchers.withId(R.id.add_meeting_room_spinner)).perform(click());
+        onView(ViewMatchers.withText(expectedMeeting.getRoom().getName())).perform(click());
+
+        for (int i = 0; i < expectedMeeting.getParticipantList().size(); i++) {
+            onView(ViewMatchers.withId(R.id.add_meeting_participant_add_EditText)).perform(replaceText(expectedMeeting.getParticipantList().get(i).getEmail()));
+            onView(ViewMatchers.withId(R.id.add_meeting_participant_add_button)).perform(click());
+        }
+
+        onView(ViewMatchers.withId(R.id.add_meeting_valid_button)).perform(click());
+
+        // Then
+        Boolean foundMeeting = false;
+        for (int i = 0; i < meetingRepository.countMeeting(); i++) {
+            if (meetingRepository.getMeetingList().get(0).getName().equals(expectedMeeting.getName())) {
+                assertEquals(expectedMeeting.getBeginDate().get(Calendar.YEAR), meetingRepository.getMeetingList().get(i).getBeginDate().get(Calendar.YEAR));
```

Comment:
hold `meetingRepository.getMeetingList().get(i)` in a variable pour simplifier la lecture